module/zfs/zfs_ioctl.c

```diff
@@ -5155,6 +5155,7 @@ static const zfs_ioc_key_t zfs_keys_recv_new[] = {
 	{"begin_record",	DATA_TYPE_BYTE_ARRAY,	0},
 	{"input_fd",		DATA_TYPE_INT32,	0},
 	{"force",		DATA_TYPE_BOOLEAN,	ZK_OPTIONAL},
+	{"heal",		DATA_TYPE_BOOLEAN,	ZK_OPTIONAL},
```

Comment:
You'll want to add this to the comment above and it should be possible to add some test coverage in `zfs-tests/cmd/libzfs_input_check/libzfs_input_check.c`.  At a minimum please document the optional argument (similar to resumable send)